src/escalus_tcp.erl

```diff
@@ -475,8 +479,59 @@ do_connect(IsSSLConnection, Address, Port, Args, SocketOpts, OnConnectFun) ->
     end,
     Reply.
 
-maybe_ssl_connection(true, Address, Port, Opts, Args) ->
-    SSLOpts = proplists:get_value(ssl_opts, Args, []),
-    ssl:connect(Address, Port, Opts ++ SSLOpts);
-maybe_ssl_connection(_, Address, Port, Opts, _) ->
-    gen_tcp:connect(Address, Port, Opts).
+maybe_ssl_connection(true, Address, Port, SocketOpts, SSLOpts) ->
+    ssl:connect(Address, Port, SocketOpts ++ SSLOpts);
+maybe_ssl_connection(_, Address, Port, SocketOpts, _) ->
+    gen_tcp:connect(Address, Port, SocketOpts).
+
+%%===================================================================
+%%% Init options parsing helpers
+%%%===================================================================
+-spec get_interface_opt([proplists:property()]) ->
+    [{ip, inet:socket_address()}].
+get_interface_opt(Opts) ->
+    case proplists:get_value(iface, Opts) of
+         undefined -> [];
+         Interface -> [{ip, iface_to_ip_address(Interface)}]
+    end.
+
+-spec get_stream_management_opt([proplists:property()]) ->
+    {boolean(), 0, inactive}.
+get_stream_management_opt(Opts) ->
+    SM = proplists:get_value(stream_management, Opts),
+    MA = proplists:get_value(manual_ack, Opts),
+    case {SM, MA} of
+        {false,_}    -> {false, 0, inactive};
+        {_, true}    -> {false, 0, inactive};
+        {true,false} -> {true, 0, inactive}
+    end.
+
+-spec overwrite_default_opts(GivenOpts :: [proplists:property()],
+                             DefaultOpts :: [proplists:property()]) ->
+    [proplists:property()].
+overwrite_default_opts([], Opts) ->
```

Comment:
Maybe `maps:merge/2`? [We require Erlang 18/19/20 for Escalus anyway](https://github.com/esl/escalus/blob/aa77ba506fd46cf2bf9a0ac3639bc2ed6e1a8173/rebar.config#L6), so there's no reason not to use maps.